### PR TITLE
implement backend framework

### PR DESCRIPTION
**Pull request by boonsboos**
from here the database API should be easy to build out.

if anyone needs help with the data processing, please let me know

### CODE REVIEW

**Comment by boonsboos** [comment]
ik heb de git history heel erg verkloot, mb

**Comment by boonsboos** [comment]
should be good though, i gave it a thourough look

just rebase and merge it

File: createDatabaseSchema.sql [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	abstract nvarchar(300) NOT NULL,
+	publicationDate date NOT NULL CONSTRAINT DF_Content_publicationDate DEFAULT GETDATE(),
+	contentStatus tinyint NOT NULL,
+
+	CONSTRAINT FK_Content_contentStatus FOREIGN KEY (contentStatus) REFERENCES contentStatus(id),
+	CONSTRAINT CK_Content_contentItemIDValid CHECK(contentItemID > 0)
+);
+
+CREATE TABLE Module (
+	contentItemID int NOT NULL,
```
Comment (FB99FIRE) [inline createDatabaseSchema.sql:63]: contentItemID moet PRIMARY KEY zijn.

File: createDatabaseSchema.sql [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	contactName nvarchar(100) NOT NULL,
+	contactEmail varchar(320) NOT NULL,
+	courseID int NOT NULL,
+
+	CONSTRAINT FK_Module_Course FOREIGN KEY (courseID) REFERENCES Course(courseID),
+	CONSTRAINT FK_Module_Content FOREIGN KEY (contentItemID) REFERENCES Content(contentItemID)
+);
+
+CREATE TABLE Webcast (
+	contentItemID int NOT NULL,
```
Comment (FB99FIRE) [inline createDatabaseSchema.sql:73]: moet nog een PRIMARY KEY achter

File: src/main/java/cgroup2/cadmycode/content/Course.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+import java.util.ArrayList;
+
+public class Course {
+    private String courseName;
+    private String subject;
+    private String introductionText;
+    private int courseID;
+    private CourseLevel level;
+    private ArrayList<Course> relatedCourses;
```
Comment (boonsboos) [inline src/main/java/cgroup2/cadmycode/content/Course.java:11]: this should not be populated by the end user or the database

**Review by FB99FIRE — APPROVED** [review]
LGTM

**Pull request merged**